javascript/code-challenges/challenges-10.test.js

```diff
@@ -0,0 +1,364 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named returnTen, takes in a string and uses split and splice to return the last 10 characters from that string as elements of an array.
+
+------------------------------------------------------------------------------------------------ */
+
+function returnTen(str){
+  return str.split("").splice(-10);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named findMax that takes in a matrix of positive numbers and returns the number with the highest value.
+
+For example:
+[
+  [1, 3, 4, 5],
+  [4, 5, 6],
+  [23, 5, 5]
+]
+
+return: 23
+------------------------------------------------------------------------------------------------ */
+const findMax = (matrix) => {
+  let temp = [];
+  Object.values(matrix).forEach(value => {
+    value.forEach( function(number) {
+      return (temp.push(number));
+    });
+  });
+  let sortmax = temp.sort(function(b, a) {
+    return a - b;
+  });
+  return sortmax[0];};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named totalSum that takes in a matrix of numbers and returns the totalSum of all the numbers.
+
+For example:
+[
+  [1, 3, 4, 5],
+  [4, 5, 1],
+  [2, 5, 5]
+]
+
+return: 35
+------------------------------------------------------------------------------------------------ */
+const totalSum = (matrix) => {
+  let temp = 0;
+  Object.values(matrix).forEach(value  => {
+    value.forEach( function(number) {
+      return temp = number+temp;
+    });
+  });
+  return temp;
+};
+
```

Comment:
So you're using the Object prototype a lot on these solutions. And while technically an array is an object in JavaScript, it's simpler not to think of it like one. This array of arrays, in most languages, is called a matrix. Try and use array methods with array objects, rather than Object prototypes.